idaes/core/surrogate/pysmo/tests/test_sampling.py

```diff
@@ -2240,5 +2241,631 @@ def test_sample_points_02(self, array_type):
             )
 
 
+class TestCustomSampling:
+    input_array = [[x, x + 10, (x + 1) ** 2 + x + 10] for x in range(10)]
+    input_array_list = [[x, x + 10, (x + 1) ** 2 + x + 10] for x in range(2)]
+    y = np.array(
+        [
+            [i, j, ((i + 1) ** 2) + ((j + 1) ** 2)]
+            for i in np.linspace(0, 10, 21)
+            for j in np.linspace(0, 10, 21)
+        ]
+    )
+    full_data = {"x1": y[:, 0], "x2": y[:, 1], "y": y[:, 2]}
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    def test__init__selection_01(self, array_type):
+        input_array = array_type(self.input_array)
+        CSClass = CustomSampling(
+            input_array,
+            number_of_samples=None,
+            sampling_type="selection",
+            list_of_distributions=["uniform", "normal"],
+        )
+        np.testing.assert_array_equal(CSClass.data, input_array)
+        np.testing.assert_array_equal(CSClass.number_of_samples, 5)
+        np.testing.assert_array_equal(CSClass.x_data, np.array(input_array)[:, :-1])
+        assert CSClass.dist_vector == ["uniform", "normal"]
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    def test__init__selection_02(self, array_type):
+        input_array = array_type(self.input_array)
+        CSClass = CustomSampling(
+            input_array,
+            number_of_samples=6,
+            sampling_type="selection",
+            list_of_distributions=["uniform", "normal"],
+        )
+        np.testing.assert_array_equal(CSClass.data, input_array)
+        np.testing.assert_array_equal(CSClass.number_of_samples, 6)
+        np.testing.assert_array_equal(CSClass.x_data, np.array(input_array)[:, :-1])
+        assert CSClass.dist_vector == ["uniform", "normal"]
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    def test__init__selection_03(self, array_type):
+        input_array = array_type(self.input_array)
+        with pytest.raises(Exception):
```

Comment:
I would recommend testing that the right `Exception` is raised here by using the `match` argument. This also helps make it clear what each test is looking for.

Reply:
Done.